src/app/ipv/middleware.ts

```diff
@@ -416,11 +420,19 @@ export const checkFormRadioButtonSelected: RequestHandler = async (
   res,
   next,
 ) => {
-  if (
-    req.body.journey === undefined &&
-    req.params.pageId !== PAGES.CHECK_MOBILE_APP_RESULT
-  ) {
+  if (req.body.journey === undefined) {
+    await handleJourneyPageRequest(req, res, next, true);
+  } else {
+    return next();
+  }
+};
+
+export const checkVcReceiptStatus: RequestHandler = async (req, res, next) => {
+  const status = await getAppVcReceipt(req);
+  if (status === AppVcReceiptStatus.PROCESSING) {
     await handleJourneyPageRequest(req, res, next, true);
+  } else if (status === AppVcReceiptStatus.ERROR) {
+    throw new Error("Failed to get VC response status");
   } else {
```

Comment:
to be safe, can we explicitly check the status is COMPLETED before allowing continue